packages/material-ui/src/Tooltip/Tooltip.js

```diff
@@ -381,6 +391,7 @@ const Tooltip = React.forwardRef(function Tooltip(props, ref) {
       children.props.onTouchEnd(event);
     }
 
+    document.body.style.WebkitUserSelect = prevUserSelect.current;
```

Comment:
Once restored we should mark the style as restored (e.g. `prevUserSelect.current = null` so that we don't accidentally restore it again if the component unmounts, mounts, and unmounts again. Any body style change applied between two unmounts would be lost. 

It's already a problem if we unmount and never had a touch interaction.

Would be easier to explain with tests. Could you add at least one and then I can expand them with some more for robustness. Though I don't think we can simulate the re-mounting yet.

Reply:
Yeah, great idea, 3 fails in a row is a good indicator it needs a test :)